moped-editor/src/views/projects/projectView/ProjectComponents/mapUtils.js

```diff
@@ -0,0 +1,256 @@
+import { useMemo } from "react";
+import { Source, Layer } from "react-map-gl";
+import { basemaps, SOURCES } from "./mapSettings";
+import { MAP_STYLES } from "./mapStyleSettings";
+
+const mapStyles = MAP_STYLES;
+
+export const useComponentFeatureCollection = (component) =>
+  useMemo(() => {
+    if (!component || !component?.features) return null;
+    return { type: "FeatureCollection", features: component.features };
+  }, [component]);
+
+/**
+ * Iterate through the mapStyles config to create an array of interactive layers
+ * @returns {Array} Array of interactive layers
+ */
+export const makeInteractiveIds = () => {
+  const interactiveLayerIds = Object.entries(mapStyles).reduce(
+    (acc, [key, value]) => {
+      if (value.isInteractive) {
+        acc.push(key);
+      }
+
+      return acc;
+    },
+    []
+  );
+
+  return interactiveLayerIds;
+};
+
+/**
+ * Component that renders Mapbox source and layers needed for the aerial basemap
+ * @param {string} basemapKey
+ * @returns JSX.Element
+ */
+export const BaseMapSourceAndLayers = ({ basemapKey }) => {
+  return (
+    <>
+      <Source {...basemaps.aerial.sources.aerials} />
+      <Layer
+        {...basemaps.aerial.layers.aerials}
+        layout={{ visibility: basemapKey === "aerial" ? "visible" : "none" }}
+      />
+      {/* Always show street labels so they can be the "target" of beforeId 
+      and always appear on top of everything else */}
+      <Layer
+        {...{
+          ...basemaps.aerial.layers.streetLabels,
+          layout: {
+            ...basemaps.aerial.layers.streetLabels.layout,
+          },
+        }}
+      />
+    </>
+  );
+};
+
+/**
+ * Component that renders all sources and layers for project components
+ * All layers are set to show below basemap street labels using beforeId = "street-labels"
+ * @param {Object} data - GeoJSON data for all project components
+ * @param {Boolean} isEditingComponent - are we editing a component?
+ * @param {String} linkMode - Tracks if we are editing "lines" or "points"
+ * @param {Object} clickedComponent - Details of the component that was clicked
+ * @param {Object} componentFeatureCollection - GeoJSON data for the component clicked
+ * @returns JSX.Element
+ */
+export const ProjectComponentsSourcesAndLayers = ({
+  data,
+  isEditingComponent,
+  linkMode,
+  clickedComponent,
+  componentFeatureCollection,
+}) => {
+  // This is a temporary to get data into the map sources
+  const {
+    ctnLinesGeojson,
+    ctnPointsGeojson,
+    projectLines,
+    projectPoints,
+    draftComponentFeatures,
+  } = data;
+
+  return (
```

Comment:
We could update this in the future to generate sources and layers from the `MAP_STYLES` config

Reply:
👍 i think it's pretty great to have it all laid out visually like this!

Reply:
My weakly held opinion here agrees with John. Until we need to export out this configuration as JSON or otherwise interact with this layout, this is super easy to read and work on.